test/mitmproxy/proxy/protocol/test_websocket.py

```diff
@@ -328,3 +334,30 @@ def test_invalid_frame(self):
         frame = websockets.Frame.from_file(self.client.rfile)
         assert frame.header.opcode == 15
         assert frame.payload == b'foobar'
+
+
+class TestStreaming(_WebSocketTest):
+
+    @classmethod
+    def handle_websockets(cls, rfile, wfile):
+        wfile.write(bytes(websockets.Frame(opcode=websockets.OPCODE.TEXT, payload=b'server-foobar')))
+        wfile.flush()
+
+    @pytest.mark.parametrize('streaming', [True, False])
+    def test_streaming(self, streaming):
+        class Stream:
+            def websocket_start(self, f):
+                f.stream = streaming
+
+        self.master.addons.add(Stream())
+        self.setup_connection()
+
+        if not streaming:
+            with pytest.raises(exceptions.TcpDisconnect):  # Reader.safe_read get nothing as result
+                frame = websockets.Frame.from_file(self.client.rfile)
+        else:
+            frame = websockets.Frame.from_file(self.client.rfile)
+
+        if streaming:
+            assert frame
+            assert self.master.state.flows[1].messages == []
```

Comment:
We should asserting *something* if we are not streaming...
Or remove the `False` parameter option.